### ISSUE TITLE
Constructor documentation has arguments in the wrong order

### ISSUE THREAD

**Issue by Groxx** [thriftrw/thriftrw-python#118]
In ipython, an error type produces this:

```python
In [11]: ErrorInfo??
Docstring:
ErrorInfo(shouldRetry=None, data=None, message=None)
The following arguments are optional:
-   shouldRetry (default: None)
-   data (default: None)
-   message (default: None)
File:      ~/.../env/local/lib/python2.7/site-packages/thriftrw/compile/scope.py
Type:      type
In [12]: ErrorInfo(False)
Out[12]: ErrorInfo({'shouldRetry': False, 'data': None, 'message': None})
In [13]: ErrorInfo(False, 'actually the message!')
Out[13]: ErrorInfo({'shouldRetry': False, 'data': None, 'message': 'actually the message!'})
In [14]: ErrorInfo(False, 'actually the message!', {'data':''})
Out[14]: ErrorInfo({'shouldRetry': False, 'data': {'data': ''}, 'message': 'actually the message!'})
```

kwargs works fine, but it's a little surprising when trying to build objects by hand for e.g. tests.

**Comment by breerly** [comment]
Interesting - good catch :+1:

**Comment by abhinav** [comment]
@Groxx Do you have the Thrift `exception` definition that produced this?

**Comment by Groxx** [comment]
@abhinav yep:
```thrift
struct ErrorInfo {
  1: optional bool shouldRetry
  2: optional string message
  3: optional map<string,string> data
}
```

**Comment by abhinav** [comment]
Okay, I just checked. The argument order is correct (it's supposed to match the order specified in the IDL but with required parameters first). The docstring is being generated incorrectly. I'll fix it.

**Comment by Groxx** [comment]
Awesome.  Thank you!

**Comment by abhinav** [comment]
The fix for this was just released in 1.2.4.
